### ISSUE TITLE
Cannot lookup symbol of nested function in method

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#6828]
_From @tinganho on January 31, 2016 17:22_

I cannot get a symbol of nested function in method.
```typescript
class A {
    b() {
       function test() {
           
       }
    }
}
```

I think the symbol `test` should show up when I do `CMD + SHIFT + O`. Since it works with nested functions:

```typescript
function test1 () {
    function test2() {
        
    }
}
```
I can get `test2` above.

_Copied from original issue: Microsoft/vscode#2584_

**Comment by dbaeumer** [comment]
_From @aeschli on February 1, 2016 10:35_

@tinganho I assume this is TypeScript?

**Comment by dbaeumer** [comment]
_From @tinganho on February 1, 2016 10:35_

Yes.

**Comment by dbaeumer** [comment]
Moving to the TS team.

**Comment by mhegazy** [comment]
this is by desing to limit noise in the list. functions are supported to allow for JS coding patterns using nested functions. which does not seem to be that common with classes.

**Comment by dbaeumer** [comment]
@mhegazy I think the tsserver nor the language service should make that 'design' decision. It should report the full structure. If a user interface does want to limit the noise we can either filter on the UI side or instruct the tsserver with an option when running the request.

**Comment by tinganho** [comment]
>this is by design to limit noise in the list.

@mhegazy I think I disagree here. It adds as much noise as if, if I would use a private method. Though the downside with a private method is that they cannot access local variables. 

I have found it quite useful to nest functions instead of declaring private methods. Since they are declared inside the methods which they are used. And most of my private methods is only used by **one** other method, I thought it was more intuitive to nest the functions instead. Though, for very large methods I got a lot of nested functions. And not having the ability to find them by symbol, makes my productivity go down. And I also think something is broken every time I don't find something auto completing or not being able to find something by symbol.

**Comment by tinganho** [comment]
Just to clarify this is the pattern I use:

Instead of this:
```typescript
if (true) {
    a += 1;
    b += 1;
    c += 1;
    d += 1;
}
```
I can refactor it as:

```typescript
if (true) {
   incrementABCD();
}

function incrementABCD() {
    a += 1;
    b += 1;
    c += 1;
    d += 1;
}
```
I've learnt this pattern through your source code :wink: . Though, I don't want to go pure functional since I think it lacks some typing that classes got.

**Comment by mhegazy** [comment]
PRs welcomed.

**Comment by DanielRosenwasser** [comment]
Also related is #4481.

**Comment by DanielRosenwasser** [comment]
Should be fixed - thanks @tinganho!
